scapy/packet.py

```diff
@@ -1407,6 +1407,56 @@ def command(self):
             c += "/" + pc
         return c
 
+    def convert_to(self, other_cls, **kwargs):
+        """Converts this Packet to another type.
+
+        This is not guaranteed to be a lossless process.
+
+        By default, this only implements conversion to ``Raw``.
+
+        :param other_cls: Reference to a Packet class to convert to.
+        :returns: Converted form of the packet, or None if no conversion is
+                  available.
+        """
+        if not issubtype(other_cls, Packet):
+            raise TypeError("{} must implement Packet".format(other_cls))
+
+        if other_cls is Raw:
+            return Raw(raw(self))
+
+        if self.haslayer(other_cls):
+            return self.getlayer(other_cls)
```

Comment:
I'm still a bit skeptical.
This behavior is strange to me :/

Reply:
That was fallback behaviour to deal with a couple of problems I had:

1. HDLC / PPP has a complex inheritance hierarchy.  I attempted to implement a PPP client in Scapy at one point, which included support for swtiching between using `HDLC`, `PPP`, and `PPP_` modes.

  I think that's more of a special case, so I'm going to leave that out.

2. One of the examples I wrote up for nRF was to extract `BTLE` packets from a `PacketList`.  I forgot to add the `PacketList.convert_to` method to this PR (which I've since added).

  I'll work around this by adding `PacketList.getlayer`, which will allow one to explicitly fetch some child layers.

I've now removed that.